exporters/metrics/src/test/java/com/google/cloud/opentelemetry/metric/MetricTranslatorTest.java

```diff
@@ -115,4 +120,38 @@ public void testMapConstantLabelWithLongValueSucceeds() {
         LabelDescriptor.newBuilder().setKey("label1").setValueType(ValueType.INT64).build();
     assertEquals(expectedLabel, actualLabel);
   }
+
+  @Test
+  public void testMapResourcesWithGCEResource() {
+    Map<AttributeKey<String>, String> testAttributes =
+        Map.of(
+            SemanticAttributes.CLOUD_PROVIDER, "gcp",
+            SemanticAttributes.CLOUD_ACCOUNT_ID, "GCE-pid",
+            SemanticAttributes.CLOUD_ZONE, "country-region-zone",
+            SemanticAttributes.CLOUD_REGION, "country-region",
+            SemanticAttributes.HOST_ID, "GCE-instance-id",
+            SemanticAttributes.HOST_NAME, "GCE-instance-name",
+            SemanticAttributes.HOST_TYPE, "GCE-instance-type");
+    AttributesBuilder attrBuilder = Attributes.builder();
+    testAttributes.forEach(
+        (key, value) -> {
+          attrBuilder.put(key, value);
+        });
+    Attributes attributes = attrBuilder.build();
+
+    Map<String, String> monitoredResourceMap =
+        MetricTranslator.mapResource(attributes, "GCE_pid").getLabelsMap();
```

Comment:
You should also add check for `monitoredResource.getType().equals("gce_instance")`